app/screens/thread_options/options/mark_as_unread_option/mark_as_unread_option.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+
+import {updateThreadRead} from '@actions/remote/thread';
+import {Screens} from '@constants';
+import {useServerUrl} from '@context/server';
+import {t} from '@i18n';
+import {dismissBottomSheet} from '@screens/navigation';
+import BaseOption from '@screens/post_options/options/base_option';
+
+import type PostModel from '@typings/database/models/servers/post';
+import type ThreadModel from '@typings/database/models/servers/thread';
+
+type Props = {
+    post: PostModel;
+    teamId: string;
+    thread: ThreadModel;
+}
+const MarkAsUnreadOption = ({teamId, thread, post}: Props) => {
+    const serverUrl = useServerUrl();
+
+    const onHandlePress = useCallback(async () => {
+        const timestamp = thread.unreadReplies ? Date.now() : post.createAt;
```

Comment:
This option is different from the `Post options`. Here we are marking the thread as unread 🙂

Reply:
So, if I understood that right, we want to use `post.createAt` instead of `post.createAt - 1` because we want all messages in the thread (after this post) to be unread. Right? That makes sense.

But I still don't get why we use `Date.now()`.

Reply:
That's for marking all replies as read. In both cases, we set the timestamp.

Reply:
Oh! I think I know where my confusion came from now. This component handles both "mark as unread" and "mark as read". For some reason I did not realize that.

Thanks!